src/util/instanceValidation.tsx

```diff
@@ -18,8 +18,12 @@ export const hasNoRootDisk = (
   return missingRoot(values.devices) && !inheritsRoot(values, profiles);
 };
 
+export const isRootDisk = (device: FormDevice): device is FormDiskDevice => {
+  return device.type === "disk" && device.path === "/" && !device.source;
```

Comment:
how do we know this is the logic for determining root disk device?

Reply:
I think we only have two types of disks: root disk and custom storage volumes. The later have a source attached and the root disk is always on the path "/". I am adding the check that the source is not set, because when adding a new custom volume a user might enter /foo but for a short time, the item would match the path and type only check.

Reply:
Looking at the [docs](https://documentation.ubuntu.com/lxd/en/latest/reference/devices_disk/#types-of-disk-devices), there are two other options device.ceph and device.cephfs All others use the device.source (except for the root storage). Will the two ceph fields to the filter as well.

Reply:
> Looking at the [docs](https://documentation.ubuntu.com/lxd/en/latest/reference/devices_disk/#types-of-disk-devices), there are two other options device.ceph and device.cephfs All others use the device.source (except for the root storage). Will the two ceph fields to the filter as well.

Actually not true, the two ceph types are also specified as the source, so this check should be fine.

Reply:
thanks for the clarification, this makes sense :+1: